ERCS/erc-7566.md

```diff
@@ -0,0 +1,193 @@
+     * @param _message is the content of the message, encoded by abi.encode.
+     * @param _messageTypes is data type array of message content.
+     * @return Message id.
+     */
+    function sendMessage(
+        uint256 _roomId,
+        uint256[] memory _to,
+        bytes memory _message,
+        Types.Type[] memory _messageTypes
+    ) external returns (uint256);
+
+    /**
+     * Get all messages received by a member in the room.
+     * @param _roomId is the id of the room.
+     * @param _memberId is the id of the member.
+     * @return An array of message ids.
+     */
+    function getMessageIds(uint256 _roomId, uint256 _memberId)
+        external
+        view
+        returns (uint256[] memory);
+
+    /**
+     * Get details of a message.
+     * @param _roomId is the id of the room.
+     * @param _messageId is the id of the message.
+     * @return The content of the message.
+     * @return Data type array of message content.
+     * @return Sender id.
+     * @return An array of receiver ids.
+     */
+    function getMessage(uint256 _roomId, uint256 _messageId)
+        external
+        view
+        returns (
+            bytes memory,
+            Types.Type[] memory,
+            uint256,
+            uint256[] memory
+        );
+}
+
+
+```
+
+### Library
+
+The library [`Types.sol`](../assets/eip-7566/Types.sol) contains an enumeration of Solidity types used in the above interfaces.
+
+## Rationale
+
+### Why are multiplayer onchain games room-based?
+
+Because the rooms are independent, each player will be assigned a new ID when entering a room. A new game round can be a room, a game task can be a room, and a game activity can be a room.
+
+### The player's state in the game.
+
+The game state refers to the player's data changes in the game, and `sendMessage` actually plays the role of a state converter. The proposal is very flexible, you can define some data inside the room (internal) or outside the room (global) according to the game logic.
+
+### How to initialize player data?
+
+You can initialize player data in `createRoom` or `joinRoom`.
+
+### How to check and handle player exits from the game?
+
+You can use `block.timestamp` or `block.number` to record the latest `sendMessage` time of a member. And add a message type to `sendMessage`. Other players can use this message type to complain that a member is offline and punish the member.
+
+### Appropriate game categories.
+
+This is a multiplayer on-chain game rather than a multiplayer real-time game standard. The game category depends on the network your contract is deployed on. Some layer 2 networks process blocks very quickly and can make some more real-time games. Generally, the network is more suitable for strategy, trading card, turn-based, chess, sandbox, and settlement.
+
+## Reference Implementation
+
+See [Multiplayer Onchain Game Example](../assets/eip-7566/MultiplayerOnchainGame.sol)
+
+## Security Considerations
+
+Needs discussion.
```

Comment:
```suggestion
<!-- TODO: Needs discussion. -->
```

If you use a comment, the linter will make sure you replace it before moving to the next status.